### PR TITLE
PUBDEV-7343: Doc improvements for Hive Import

### PR DESCRIPTION
**Pull request by angela0xdata**
- Noted that Direct metatore access is the preferred method for import Hive tables over JDBC
- Updated examples
- prettified code using code-block:: bash instead of ::

### CODE REVIEW

File: h2o-docs/src/product/getting-data-into-h2o.rst [hunk 1]
```diff
@@ -333,42 +395,118 @@ The ``import_sql_select`` function accepts the following parameters:
...
-- For Hortonworks, Hive JDBC client jars can be found on one of the edge nodes after you have installed HDP: ``/usr/hdp/current/hive-client/lib/hive-jdbc-<version>-standalone.jar``. More information is available here: `https://docs.hortonworks.com/HDPDocuments/HDP2/HDP-2.6.4/bk_data-access/content/hive-jdbc-odbc-drivers.html <https://docs.hortonworks.com/HDPDocuments/HDP2/HDP-2.6.4/bk_data-access/content/hive-jdbc-odbc-drivers.html>`__
-- For Cloudera, install the JDBC package for your operating system, and then add ``/usr/lib/hive/lib/hive-jdbc-<version>-standalone.jar`` to your classpath. More information is available here: `https://www.cloudera.com/documentation/enterprise/5-3-x/topics/cdh_ig_hive_jdbc_install.html <https://www.cloudera.com/documentation/enterprise/5-3-x/topics/cdh_ig_hive_jdbc_install.html>`__
-- You can also retrieve this from Maven for the desire version using ``mvn dependency:get -Dartifact=groupId:artifactId:version``.
+       beeline -u jdbc:hive2://hive-host:10000/db-name
 
-**Provide H2O with the JDBC Driver**
+    c. Create the DB table:
 
-Add the Hive JDBC driver to H2O's classpath:
+     .. code-block:: bash
```
Comment (honzasterba) [inline h2o-docs/src/product/getting-data-into-h2o.rst:414]: this is not bash but SQL

File: h2o-docs/src/product/getting-data-into-h2o.rst [hunk 2]
```diff
@@ -333,42 +395,118 @@ The ``import_sql_select`` function accepts the following parameters:
...
+           ROW FORMAT DELIMITED
+           FIELDS TERMINATED BY ','
+           LOCATION '/tmp';
 
-Start the h2o.jar in the terminal with your downloaded JDBC driver in the classpath: 
+    d. Import the data from the dataset: 
 
-.. tabs::
-   .. code-tab:: r R
+    .. code-block:: bash
```
Comment (honzasterba) [inline h2o-docs/src/product/getting-data-into-h2o.rst:437]: this is SQL, not bash nor R

File: h2o-docs/src/product/getting-data-into-h2o.rst [hunk 3]
```diff
@@ -333,42 +395,118 @@ The ``import_sql_select`` function accepts the following parameters:
...
 
-        h2o.init(extra_classpath = "hive-jdbc.jar")
+      LOAD DATA INPATH '/tmp/AirlinesTest.csv' OVERWRITE INTO TABLE AirlinesTest
```
Comment (honzasterba) [inline h2o-docs/src/product/getting-data-into-h2o.rst:439]: should be stated that the file has to be present on HDFS in /tmp

File: h2o-docs/src/product/getting-data-into-h2o.rst [hunk 4]
```diff
@@ -190,52 +192,109 @@ core-site.xml must be configured for at least the following properties (class, p
...
+''''''''''''''''''
+
+Example 1: Access Metadata via Metastore (preferred)
+####################################################
 
 .. tabs::
+
    .. code-tab:: r R
 
+        # Add the Hive JDBC driver to H2O's classpath
```
Comment (honzasterba) [inline h2o-docs/src/product/getting-data-into-h2o.rst:226]: maybe we should split this into bash code block and R part as well?

File: h2o-docs/src/product/getting-data-into-h2o.rst [hunk 5]
```diff
@@ -190,52 +192,109 @@ core-site.xml must be configured for at least the following properties (class, p
...
+        java -cp hive-jdbc.jar:<path_to_h2o_jar>: water.H2OApp
```
Comment (honzasterba) [inline h2o-docs/src/product/getting-data-into-h2o.rst:227]: also here h2o needs to be started on hadoop with hadoop jar command

File: h2o-docs/src/product/getting-data-into-h2o.rst [hunk 6]
```diff
@@ -190,52 +192,109 @@ core-site.xml must be configured for at least the following properties (class, p
...
-        basic_import <- h2o.import_hive_table("jdbc:hive2://hive-server:10000/default", "table_name")
-        # access metadata via Metastore
-        multi_format_enabled <- h2o.import_hive_table("default", "table_name", allow_multi_format=True)
-        with_partition_filter <- h2o.import_hive_table("default", "table_name", [["2017", "02"]])
+        # multi-format import
+        multi_format_enabled <- h2o.import_hive_table("default", 
+                                                      "table_name", 
+                                                      allow_multi_format=True)
 
+        # import with partitioned filter
```
Comment (honzasterba) [inline h2o-docs/src/product/getting-data-into-h2o.rst:241]: this filter actual filters which partitions are imported so I would call it a "partition filter" (not partitioned)

File: h2o-docs/src/product/getting-data-into-h2o.rst [hunk 7]
```diff
@@ -190,52 +192,109 @@ core-site.xml must be configured for at least the following properties (class, p
...
+        with_partition_filter <- h2o.import_hive_table("default", 
+                                                       "table_name", 
+                                                       [["2017", "02"]])
 
    .. code-tab:: python
 
+        # Add the Hive JDBC driver to H2O's classpath
+        java -cp hive-jdbc.jar:<path_to_h2o_jar>: water.H2OApp
```
Comment (honzasterba) [inline h2o-docs/src/product/getting-data-into-h2o.rst:249]: same here would be started with hadoop jar

File: h2o-docs/src/product/getting-data-into-h2o.rst [hunk 8]
```diff
@@ -190,52 +192,109 @@ core-site.xml must be configured for at least the following properties (class, p
...
+
+        # start the h2o.jar in Python
```
Comment (honzasterba) [inline h2o-docs/src/product/getting-data-into-h2o.rst:251]: its not possible to start h2o on hadoop from py/r

**Comment by angela0xdata** [comment]
@honzasterba I think I captured your comments. I have a couple questions, though.

1. Which is the correct format for copying the Hive JDBC driver to H2O’s classpath:
- `java -cp hive-jdbc.jar:<path_to_h2o_jar> water.H2OApp`
- `java -cp hive-jdbc.jar:<path_to_h2o_jar>: water.H2OApp`

2. For [Direct Hive Import](http://docs.h2o.ai/h2o/latest-stable/h2o-docs/getting-data-into-h2o.html#direct-hive-import), do you need to initialize H2O with extra_classpath?
`h2o.init(extra_classpath=["hive-jdbc-standalone.jar"])`

**Comment by honzasterba** [comment]
1) java -cp hive-jdbc.jar:<path_to_h2o_jar> water.H2OApp
2) for direct hive-import h2o needs to be running on hadoop so its not started with h2o.init

File: h2o-docs/src/product/getting-data-into-h2o.rst [hunk 9]
```diff
@@ -190,52 +192,135 @@ core-site.xml must be configured for at least the following properties (class, p
     </configuration>
 
 
-Direct Hive import
+Direct Hive Import
 ~~~~~~~~~~~~~~~~~~
 
-H2O supports direct ingestion of data managed by Hive in Hadoop. This feature is available only when H2O is running as a Hadoop job. Internally H2O uses metadata in Hive Metastore database to determine the location and format of given Hive table. H2O then imports data directly from HDFS so limitations of supported formats mentioned above apply. Data from hive can pulled into H2O using ``import_hive_table`` function. H2O can read Hive table metadata two ways - either via direct Metastore access or via JDBC.
+H2O supports direct ingestion of data managed by Hive in Hadoop. This feature is available only when H2O is running as a Hadoop job. Internally H2O uses metadata in Hive Metastore database to determine the location and format of given Hive table. H2O then imports data directly from HDFS so limitations of supported formats mentioned above apply. Data from hive can pulled into H2O using ``import_hive_table`` function. H2O can read Hive table metadata two ways - either via direct Metastore access (preferred) or via JDBC.
```
Comment (honzasterba) [inline h2o-docs/src/product/getting-data-into-h2o.rst:198]: actually I dont think either way of reading hive metadata is preffered, they both work well

maybe you have this mistaken with the Direct hive import being prefered over reading data from Hive via JDBC

Comment (honzasterba) [reply h2o-docs/src/product/getting-data-into-h2o.rst:198]: and the JDBC access to metadata is actually easier to setup

File: h2o-docs/src/product/getting-data-into-h2o.rst [hunk 10]
```diff
@@ -190,52 +192,135 @@ core-site.xml must be configured for at least the following properties (class, p
...
 
-Requirements:
+Requirements
+''''''''''''
 
-- Direct Metastore access - Hive jars and configuration must be present on H2O job classpath - either via adding it to yarn.application.classpath (or similar property for your resource manger of choice) or by adding Hive jars and configuration to libjars.
-- JDBC metadata access - Hive JDBC Driver must be on H2O job classpath. (See below for details about access Kerberized Hive via JDBC.)
 - The user running H2O must have read access to Hive and the files it manages.
 
+- For Direct Metastore access, the Hive jars and configuration must be present on H2O job classpath - either by adding it to yarn.application.classpath (or similar property for your resource manger of choice) or by adding Hive jars and configuration to libjars. Note that Direct Metastore access is the perferred method.
```
Comment (honzasterba) [inline h2o-docs/src/product/getting-data-into-h2o.rst:205]: same as above

File: h2o-docs/src/product/getting-data-into-h2o.rst [hunk 11]
```diff
@@ -190,52 +192,135 @@ core-site.xml must be configured for at least the following properties (class, p
...
+####################################################
+
+This example shows how to access metadata via the metastore. 
+
+1. Add the Hive JDBC driver to H2O's classpath.
+
+ .. code-block:: bash
+
+      # Add the Hive JDBC driver to H2O's classpath
+      java -cp hive-jdbc.jar:<path_to_h2o_jar> water.H2OApp
```
Comment (honzasterba) [inline h2o-docs/src/product/getting-data-into-h2o.rst:229]: for direct hive import h2o needs to be running on hadoop so this does not apply, only the paragraph 2. below

File: h2o-docs/src/product/getting-data-into-h2o.rst (hunk 10 above)
Comment (angela0xdata) [reply h2o-docs/src/product/getting-data-into-h2o.rst:205]: So now I'm confused about the Jira ticket: https://0xdata.atlassian.net/browse/PUBDEV-7343
> Some people end-up using JDBC interface which is known to be sub-optimal.
> We need to make sure there is a clearly preferred method, with a clear description...

Comment (honzasterba) [reply h2o-docs/src/product/getting-data-into-h2o.rst:205]: the ticket is about using JDBC http://docs.h2o.ai/h2o/latest-stable/h2o-docs/getting-data-into-h2o.html#jdbc-databases vs using filesystem or direct hive import http://docs.h2o.ai/h2o/latest-stable/h2o-docs/getting-data-into-h2o.html#direct-hive-import

**Comment by angela0xdata** [comment]
@honzasterba thank you! You're review and comments were so helpful!

**Pull request merged**